include/aie/Dialect/AIE/IR/AIETargetModel.h

```diff
@@ -232,9 +246,8 @@ class AIETargetModel {
   // Run consistency checks on the target model.
   void validate() const;
 
-  // Return true if this is an NPU-based device
-  // There are several special cases for handling the NPU at the moment.
-  virtual bool isNPU() const { return false; }
+  // Return true if this device has a given property.
+  virtual bool hasProperty(ModelProperty Prop) const = 0;
```

Comment:
```suggestion
  bool hasProperty(ModelProperty Prop) const { return false; }
```

But it might also be worth looking at how rtti works in AIETargetModel. Using a similar scheme, each subclass would add it's bitvector of features via the call to the base class constructor. The base class would OR the bitvector with it's own, so the subclass only needs to add new features not present in the base class.
```c++
#include <iostream>
#include <cstdlib>

class ClassA {
private:
  const uint32_t features;
public:
  ClassA(uint32_t f) : features(f) {}
  uint32_t getFeatures() const { return features; };
};

class ClassB : public ClassA {
  static constexpr uint32_t class_b_features = 0x1;
public:
  ClassB(uint32_t f) : ClassA(f | class_b_features) {}
  ClassB() : ClassA(class_b_features) {}
};

class ClassC : public ClassB {
  static constexpr uint32_t class_c_features = 0x2;
public:
  ClassC() : ClassB(class_c_features) {}
};

class ClassD : public ClassB {
  static constexpr uint32_t class_d_features = 0x4;
public:
  ClassD() : ClassB(class_d_features ) {}
};

int main(int argc, char *argv[]) {
  std::cout << "B " << ClassB().getFeatures() << "\n";
  std::cout << "C " << ClassC().getFeatures() << "\n";
  std::cout << "D " << ClassD().getFeatures() << "\n";
}
```
```bash
$ ./a.out
B 1
C 3
D 5
```

Reply:
Good point. PR updated. Thanks!